src/tools/generators.ts

```diff
@@ -601,3 +608,275 @@ export async function generateAppIcons(option: `${Platforms}` | "all") {
+    androidSize <= 0 && "   • a value greater than 0",
+    androidSize >= 288 && "   • a value less than 288",
+  ].filter(Boolean)
+
+  if (androidMessages.length) {
+    validationMessages.push(`⚠️  "--android-size" option must be:`, ...androidMessages)
+  }
+
+  // check if the ios size option is numerical and non-zero
+  const iosMessages = [
+    Number.isNaN(iosSize) && "   • a numerical value",
+    iosSize <= 0 && "   • a value greater than 0",
+  ].filter(Boolean)
+
+  if (iosMessages.length) {
+    validationMessages.push(`⚠️  "--ios-size" option must be:`, ...iosMessages)
+  }
+
+  // check if the background option is a valid hex color
+  if (!/^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(backgroundColor)) {
+    validationMessages.push(`⚠️  background color parameter must be:`)
+    validationMessages.push(`   • a valid hex color`)
+  }
+
+  // validate template input file
+  const boilerplateInputFilePath = path(sourceDirectory(), "splash-screen", "logo.png")
+  const inputFilePath = path(templatesDir(), "splash-screen", "logo.png")
+
+  const isMissing = !exists(inputFilePath)
+  const isInvalidSize = await (async function () {
+    if (isMissing) return false
+
+    const metadata = await sharp(inputFilePath).metadata()
+    return metadata.width !== 1024 || metadata.height !== 1024
+  })()
+  const isSameAsSource = await (async function () {
+    if (skipSourceEqualityValidation) return false
+    if (isMissing) return false
+
+    const inputFileMd5 = inspect(inputFilePath, { checksum: "md5" }).md5
+    const sourceFileMd5 = inspect(boilerplateInputFilePath, { checksum: "md5" }).md5
+
+    return inputFileMd5 === sourceFileMd5
+  })()
+
+  const messages = [
+    isMissing && "   • the file is missing",
+    isInvalidSize && "   • the file is the wrong size (expected 1024x1024px)",
+    isSameAsSource &&
+      "   • looks like you're using our default template; customize the file with your own icon first",
+  ].filter(Boolean)
+
+  if (messages.length) {
+    validationMessages.push(`⚠️  ignite/templates/splash-screen/logo.png:`, ...messages)
+  }
+
+  return {
+    isValid: !validationMessages.length,
+    messages: validationMessages,
+  }
+}
+
+/**
+ * Generates splash screen for all platforms
+ */
+export async function generateSplashScreen(options: {
+  androidSize: number
+  iosSize: number
+  backgroundColor: string
+}) {
+  const { androidSize, iosSize, backgroundColor } = options || {}
+  const { path, exists, write, find } = filesystem
+  const cwd = process.cwd()
+
+  const inputFilePath = path(templatesDir(), "splash-screen", "logo.png")
+  const expoOutputDirPath = path(cwd, "assets/images")
+  const isExpoOutputDirExists = exists(expoOutputDirPath) === "dir"
+  const bootsplashCliPath = path(
+    cwd,
+    "node_modules/react-native-bootsplash/dist/commonjs/generate.js",
+  )
+  const isBootsplashCliInstalled = exists(bootsplashCliPath) === "file"
+  const optionGenerationSuccesses = []
+
```

Comment:
below, we have 2 different methods for generating splash-screen assets.

for Vanilla, we use `react-native-bootsplash` CLI (consumed directly by importing the script file)

for Expo, we use the already-installed Sharp library.